src/ring_theory/localization.lean

```diff
@@ -723,6 +797,54 @@ def le_rel_embedding :
   map_rel_iff'   := λ J₁ J₂, ⟨ideal.comap_mono, λ hJ,
     f.map_comap J₁ ▸ f.map_comap J₂ ▸ ideal.map_mono hJ⟩ }
 
+/-- If `R` is a ring, then prime ideals in the localization at `M`
+correspond to prime ideals in the original ring `R` that are disjoint from `M`.
+This gives the particular case for an ideal and its comap,
+see `le_rel_iso_of_prime` for the more general relation isomorphism -/
+lemma is_prime_iff_is_prime_disjoint (J : ideal S) :
+  J.is_prime ↔ (ideal.comap f.to_map J).is_prime ∧ disjoint (M : set R) ↑(ideal.comap f.to_map J) :=
+begin
+  split,
+  { refine λ h, ⟨⟨_, _⟩, λ m hm, h.1 (ideal.eq_top_of_is_unit_mem _ hm.2 (map_units f ⟨m, hm.left⟩))⟩,
+    { refine λ hJ, h.left _,
+      rw [eq_top_iff, (le_rel_embedding f).map_rel_iff],
+      exact le_of_eq hJ.symm },
+    { intros x y hxy,
+      rw [ideal.mem_comap, ring_hom.map_mul] at hxy,
+      exact h.right hxy } },
+  { refine λ h, ⟨λ hJ, h.left.left (eq_top_iff.2 _), _⟩,
+    { rwa [eq_top_iff, (le_rel_embedding f).map_rel_iff] at hJ },
+    { intros x y hxy,
+      obtain ⟨a, s, ha⟩ := mk'_surjective f x,
+      obtain ⟨b, t, hb⟩ := mk'_surjective f y,
+      have : f.mk' (a * b) (s * t) ∈ J := by rwa [mk'_mul, ha, hb],
+      rw [mk'_mem_iff, ← ideal.mem_comap] at this,
+      replace this := h.left.right this,
+      rw [ideal.mem_comap, ideal.mem_comap] at this,
+      rwa [← ha, ← hb, mk'_mem_iff, mk'_mem_iff] } }
+end
+
+/-- If `R` is a ring, then prime ideals in the localization at `M`
+correspond to prime ideals in the original ring `R` that are disjoint from `M`.
+This gives the particular case for an ideal and its map,
+see `le_rel_iso_of_prime` for the more general relation isomorphism, and the reverse implication -/
+lemma is_prime_of_is_prime_disjoint (I : ideal R) :
+  I.is_prime ∧ disjoint (M : set R) ↑I → (ideal.map f.to_map I).is_prime :=
+λ h, by rwa [is_prime_iff_is_prime_disjoint f, comap_map_of_is_prime_disjoint f I h.1 h.2]
```

Comment:
Usually a lemma doesn't have a hypothesis `p ∧ q`, but hypotheses `p` and `q` separately:
```suggestion
lemma is_prime_of_is_prime_disjoint (I : ideal R)
  (hp : I.is_prime) (hd : disjoint (M : set R) ↑I) : (ideal.map f.to_map I).is_prime :=
begin
  rw [is_prime_iff_is_prime_disjoint f, comap_map_of_is_prime_disjoint f I hp hd],
  exact ⟨hp, hd⟩
end
```

Reply:
I'd be fine changing this, but as it is it has a symmetry with `is_prime_iff_is_prime_disjoint ` and `le_rel_iso_of_prime ` that makes a lot of the proofs cleaner (and makes it easier to use them together without de-structuring conjunctions each time).

Reply:
@Vierkantor @jcommelin I've fixed everything except this now, so I guess let me know if you'd like me to go through with making this change or not.

Reply:
I agree with @Vierkantor. Locally, it might make it nicer to work with `\and`. But globally, it's better to have separate hypotheses.

Reply:
Okay, sounds good, I've made this change then. I also made an identical change to `is_maximal_of_is_maximal_disjoint` which is the maximal ideal version of this same statement. Let me know if there are any other changes to make now.